tests/lib/rules/html-indent.js

```diff
@@ -0,0 +1,4214 @@
+  const lineIndents = lines.filter(line => line.trim()).map(line => line.match(/ */)[0].length)
+  const minLineIndent = Math.min.apply(null, lineIndents)
+
+  return lines.map(line => line.slice(minLineIndent)).join('\n')
+}
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const tester = new RuleTester({
+  parser: 'vue-eslint-parser',
+  parserOptions: {
+    ecmaVersion: 2017,
+    ecmaFeatures: {
+      globalReturn: true
+    }
+  }
+})
+
+tester.run('html-indent', rule, {
+  valid: [
+    // Examples in docs.
+    unIndent`
+      <template>
+          <div class="foo">
+              Hello.
+          </div>
+          <div
+              id="a"
+              class="b"
+              :other-attr="{
+                  aaa: 1,
+                  bbb: 2
+              }"
+              @other-attr2="
+                  foo();
+                  bar();
+              "
+          >
+              {{
+                  displayMessage
+              }}
+          </div>
+      </template>
+    `,
+
+    // VAttribute
+    unIndent`
+      <template>
+          <div
+              a="a"
+              b="b"
+              c=
+                  "c"
+              d
+                  ="d"
+              e
+              f
+                  =
+          ></div>
+      </template>
+    `,
+    unIndent`
+      <template>
+          <div a="a"
+               b="b"
+               c=
+                   "c"
+               d
+                   ="d"
+               e
+               f
+                   =
+          ></div>
+      </template>
+    `,
+    unIndent`
+      <template>
+          <div a="
+          a" b="b"></div>
+      </template>
+    `,
+
+    // VExpressionContainer
+    unIndent`
+      <template>
+          <div
+              :foo="
+                  value
+              "
+          ></div>
+      </template>
+    `,
+
+    // VForExpression
+    unIndent`
+      <template>
+          <div
+              v-for="
+                  x
+                      in
+                          xs
+              "
+          ></div>
+          <div
+              v-for="
+                  (
+                      x
+                      ,
+                      y
+                      ,
+                      z
+                  )
+                      of
+                          xs
+              "
+          ></div>
+      </template>
+    `,
+
+    // VOnExpression
+    unIndent`
+      <template>
+          <div
+              v-on:a="
+                  foo(); bar();
+              "
+              v-on:b="
+                  foo()
+                  bar()
+              "
+              v-on:c="foo()
+                      bar()"
+          ></div>
+      </template>
+    `,
+
+    // VStartTag
+    unIndent`
+      <template>
+          <div
+              aaa
+              bbb
+    `,
```

Comment:
not a valid template, is is intentional?

and why the parser not reporting an error?

Reply:
Yes, it's intentional. The HTML spec includes recovering process for invalid syntax, so `vue-eslint-parser` reports syntax errors with the `programNode.templateBody.errors` property instead of throwing the error. Then [vue/no-parsing-error](https://github.com/vuejs/eslint-plugin-vue/blob/94f387ed68149f8d12c1960d68cb5d9c376b9a89/docs/rules/no-parsing-error.md) rule would report those syntax errors.

The purpose of this test case is to check whether `vue/html-indent` rule works properly on an incomplete code. Especially, the expected indentation of closing brackets (`>`) does not apply to the last attribute even if the closing bracket is lacking.